model/task/task.go

```diff
@@ -2800,88 +2815,91 @@ func GetTasksByVersion(versionID string, sortBy []TasksSortOrder, statuses []str
 					tempParentKey: []interface{}{},
 				},
 			},
-			// get any annotation that has at least one issue
-			{
-				"$lookup": bson.M{
-					"from": annotations.Collection,
-					"let":  bson.M{"task_annotation_id": "$" + IdKey, "task_annotation_execution": "$" + ExecutionKey},
-					"pipeline": []bson.M{
-						{
-							"$match": bson.M{
-								"$expr": bson.M{
-									"$and": []bson.M{
-										{
-											"$eq": []string{"$" + annotations.TaskIdKey, "$$task_annotation_id"},
-										},
-										{
-											"$eq": []string{"$" + annotations.TaskExecutionKey, "$$task_annotation_execution"},
-										},
-										{
-											"$ne": []interface{}{
-												bson.M{
-													"$size": bson.M{"$ifNull": []interface{}{"$" + annotations.IssuesKey, []bson.M{}}},
-												}, 0,
-											},
+		}...)
+	}
+	pipeline = append(pipeline, []bson.M{
```

Comment:
should we make annotation getting optional somehow? for example we don't need it here.

Reply:
We would need it any time the task is displayed to the user and they are able to add an annotation (which will eventually be every task). This will be surfaced to the user in some table of commits, so we probably do need it.

Reply:
But in this case we're doing a mutation so it's not about displaying data, it's about changing data, right? So for this mutation we don't use annotations.

Reply:
It looks like the mutation is returning tasks. But I'm not sure, it would also depend on how it's being used on the front end. Mohamed will know better.

Reply:
So this mutation does return tasks it allows us to take advantage of apollo and automatically update the state of the tasks in the UI with the latest statuses returned from the mutation response. 

Technically it make sense to process the tasks the same way as they get processed elsewhere in the UI. But practically we expect all these tasks to return "Will run" so we might be able to get away with omitting the annotation getting. Do y'all think its worth it to remove it? I don't have any strong opinions either way.

Reply:
When I compared how long the graphql query takes with and with the annotation lookup, it didn't seem like a significant increase, so I don't think leaving it there is detrimental. But I will leave the ultimate decision to Annie.

Reply:
I'm fine with leaving it as is so we can get this PR through; if we need to do a speed up at some point we can revisit.